features/environment.py

```diff
@@ -1,17 +1,63 @@
 """Behave environment setup commands"""
+from __future__ import annotations
 
 import os
+import shlex
 import shutil
 import subprocess
 import tempfile
 import venv
 from pathlib import Path
-
-from typing import Set
+from typing import Any, Set
 
 _PATHS_TO_REMOVE: Set[Path] = set()
 
 
+
+
+def run(
+    cmd: list | str, split: bool = True, print_output: bool = False, **kwargs: Any
+) -> subprocess.CompletedProcess:
+    """Run a shell command.
+
+    Args:
+        cmd: A command string, or a command followed by program
+            arguments that will be submitted to Popen to run.
+
+        split: Flag that splits command to provide as multiple *args
+            to Popen. Default is True.
+
+        print_output: If True will print previously captured stdout.
+            Default is False.
+
+        **kwargs: Extra options to pass to subprocess.
+
+    Example:
+    ::
+        "ls"
+        "ls -la"
+        "chmod 754 local/file"
+
+    Returns:
+        Result with attributes args, returncode, stdout and stderr.
+
+    """
+    if isinstance(cmd, str) and split:
+        cmd = shlex.split(cmd)
+    result = subprocess.run(cmd, input="", capture_output=True, **kwargs)  # noqa: PLW1510
+    result.stdout = result.stdout.decode("utf-8")
+    result.stderr = result.stderr.decode("utf-8")
+    if print_output:
+        print(result.stdout)
+    return result
+
+def call(cmd, env):
+    res = run(cmd, env=env)
+    if res.returncode:
+        print(res.stdout)
+        print(res.stderr)
+        assert False
```

Comment:
Is this more or less `check_output`? https://docs.python.org/3/library/subprocess.html#subprocess.check_output

Reply:
This chunk of code is borrowed from Kedro repo.